# Module 1 Lab 3: Textbook Problem 1.2
**Name:** Ramisha Tasfia  


In [1]:
import pandas as pd

# Raw ratings from textbook table
data = {
    'provider': ['GEICO', 'GEICO', 'GEICO',
                 'Progressive', 'Progressive', 'Progressive',
                 'USAA', 'USAA', 'USAA'],
    'rating': [4.7, 8.3, 9.2,
               7.4, 6.7, 8.9,
               3.8, 6.3, 8.1]
}

df = pd.DataFrame(data)
print(df.to_string(index=False))

   provider  rating
      GEICO     4.7
      GEICO     8.3
      GEICO     9.2
Progressive     7.4
Progressive     6.7
Progressive     8.9
       USAA     3.8
       USAA     6.3
       USAA     8.1


## Summary Statistics: Min, Mean, Max per Provider

In [2]:
summary = df.groupby('provider')['rating'].agg(
    Min='min',
    Mean='mean',
    Max='max'
).round(2)

print(summary)

             Min  Mean  Max
provider                   
GEICO        4.7  7.40  9.2
Progressive  6.7  7.67  8.9
USAA         3.8  6.07  8.1


## Recommendation

The summary table shows:

| Provider    | Min | Mean | Max |
|-------------|-----|------|-----|
| GEICO       | 4.7 | 7.40 | 9.2 |
| Progressive | 6.7 | 7.67 | 8.9 |
| USAA        | 3.8 | 6.07 | 8.1 |

**I would choose Progressive.**

GEICO and Progressive both have an almost identical mean (GEICO-7.40 and Progressive-7.67) and therefore it isn't immediately apparent based solely on their means which would be the best provider. However, the minimal ratings are very different. The low for GEICO is 4.7, indicating at least one customer suffered greatly with this service. The low for Progressive is 6.7, a respectable number. The low for USAA is 3.8, thus we can easily rule it out.

This makes sense because in looking at the minimum, we see how poorly the least fortunate customer was treated by each provider. Insurance providers are expected to offer consistent service regardless if they meet or exceed expectations. Therefore, the provider that has the highest mean will also likely be the safest. In addition to being safer, the higher performing option will provide a greater value. As such, Progressive was clearly the best provider to choose in my opinion. 